contribs/dvrp/src/main/java/org/matsim/contrib/util/stats/VehicleOccupancyProfileWriter.java

```diff
@@ -160,4 +168,31 @@ private String filename(String prefix) {
 		return matsimServices.getControlerIO()
 				.getIterationFilename(matsimServices.getIterationNumber(), prefix + "_" + mode);
 	}
+
+	private String outputFilename(String prefix) {
+		return matsimServices.getControlerIO()
+				.getOutputFilenameWithOutputPrefix(prefix + "_" + mode);
+	}
+
+	@Override
+	public void notifyShutdown(ShutdownEvent event) {
+		dumpOutput(outputFile, ".txt");
+		dumpOutput(outputFile + "_" + TimeProfileCharts.ChartType.Line, ".png");
+		dumpOutput(outputFile + "_" + TimeProfileCharts.ChartType.StackedArea, ".png");
+	}
+
+	private void dumpOutput(String prefix, String extension) {
+		try {
+			File toFile = new File(outputFilename(prefix) + extension);
+			File fromFile = new File(filename(prefix) + extension);
+			try {
+				Files.copy(fromFile.toPath(), toFile.toPath(), StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.COPY_ATTRIBUTES);
+			} catch (IOException e) {
+				throw new UncheckedIOException(e);
+			}
+		} catch (Exception ee) {
+			LogManager.getLogger(this.getClass()).error("writing output " + outputFilename(prefix + extension) +
+					"did not work; probably parameters were such that no such output was generated in the final iteration");
+		}
+	}
```

Comment:
This looks like a method that could be called from 100 other places. Maybe we have a util for it?